spec/02-integration/05-proxy/21-reports_spec.lua

```diff
@@ -223,16 +367,73 @@ for _, strategy in helpers.each_strategy() do
                                        ":" .. helpers.get_proxy_port(true) .. "/up-ws")
 
       reports_send_ping()
-      local _, reports_data = assert(reports_server:stop())
 
+      local _, reports_data = assert(reports_server:stop())
       assert.same(1, #reports_data)
       assert.match("requests=1", reports_data[1])
       assert.match("http_reqs=0", reports_data[1])
       assert.match("https_reqs=0", reports_data[1])
+      assert.match("h2c_reqs=0", reports_data[1])
+      assert.match("h2_reqs=0", reports_data[1])
       assert.match("grpc_reqs=0", reports_data[1])
       assert.match("grpcs_reqs=0", reports_data[1])
       assert.match("ws_reqs=0", reports_data[1])
       assert.match("wss_reqs=1", reports_data[1])
     end)
+
+    it("#stream reports tcp streams", function()
+      local tcp = require "socket".tcp()
+      assert(tcp:connect(helpers.get_proxy_ip(false), 19000))
+
+      -- TODO: we need to get rid of the next line!
+      assert(tcp:send(MESSAGE))
+
+      local body = assert(tcp:receive("*a"))
+      assert.equal(MESSAGE, body)
+
+      tcp:close()
+
+      reports_send_stream_ping()
+
+      local _, reports_data = assert(reports_server:stop())
+      assert.same(1, #reports_data)
+      assert.match("streams=1", reports_data[1])
+      assert.match("tcp_streams=1", reports_data[1])
+      assert.match("tls_streams=0", reports_data[1])
+    end)
+
+    it("#stream reports tls streams", function()
+      local tcp = require "socket".tcp()
+      local ssl = require("ssl")
+
+      assert(tcp:connect(helpers.get_proxy_ip(true), 19443))
+
+      tcp = ssl.wrap(tcp, {
+        mode     = "client",
+        verify   = "none",
+        protocol = "any",
+      })
+
+      -- TODO: should SNI really be mandatory?
+      tcp:sni("this-is-needed.test")
+
+      assert(tcp:dohandshake())
+
+      -- TODO: we need to get rid of the next line!
```

Comment:
Same: what does "next line" mean here

Reply:
It means that we should not need it, e.g. the test should work even when that line is removed. That line is there to please our "restricted" stream support.